orchestration/dagster_orchestration/hca_orchestration/pipelines/load_hca.py

```diff
@@ -72,17 +83,96 @@
+        }
+    ]
+
+
+@success_hook(
+    required_resource_keys={'slack', 'target_hca_dataset', 'dagit_config'}
+)
+def import_start_notification(context: HookContext) -> None:
+    context.log.info(f"Solid output = {context.solid_output_values}")
+    job_id = context.solid_output_values["result"]
+    kvs = {
+        "Staging Area": context.solids.pre_process_metadata.input_prefix,
+        "Target Dataset": context.resources.target_hca_dataset.dataset_name,
+        "Jade Project": context.resources.target_hca_dataset.project_id,
+        "TDR Job ID": job_id,
+        "Dagit link": f'<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>'
     }
+
+    context.resources.slack.send_message(blocks=_base_slack_blocks("HCA Starting Import", key_values=kvs))
+
+
+@failure_hook(
+    required_resource_keys={'slack', 'target_hca_dataset', 'dagit_config'}
+)
+def import_failed_notification(context: HookContext) -> None:
+    if "result" in context.solid_output_values:
+        job_id = context.solid_output_values["result"]
+    else:
+        job_id = "N/A"
+
+    kvs = {
+        "Staging Area": context.solids.pre_process_metadata.input_prefix,
+        "Target Dataset": context.resources.target_hca_dataset.dataset_name,
+        "Jade Project": context.resources.target_hca_dataset.project_id,
+        "TDR Job ID": job_id,
+        "Dagit link": f'<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>'
+    }
+
+    context.resources.slack.send_message(blocks=_base_slack_blocks("HCA Import Failed", key_values=kvs))
+
+
+@success_hook(
+    required_resource_keys={'slack', 'target_hca_dataset', 'dagit_config'}
 )
+def message_for_import_done(context: HookContext) -> None:
+    context.resources.slack.send_message(blocks=_base_slack_blocks("HCA Import Complete", {
+        "Staging Area": context.solids.pre_process_metadata.input_prefix,
+        "Target Dataset": context.resources.target_hca_dataset.dataset_name,
+        "Jade Project": context.resources.target_hca_dataset.project_id,
+        "Dagit link": f'<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>'
+    }))
 
 
 @pipeline(
     mode_defs=[prod_mode, dev_mode, local_mode, test_mode]
 )
 def load_hca() -> None:
-    staging_dataset = create_scratch_dataset(pre_process_metadata(clear_scratch_dir()))
-    result = import_data_files(staging_dataset).collect()
+    hooked_staging_dataset = create_scratch_dataset(pre_process_metadata(clear_scratch_dir())).with_hooks({import_start_notification})
+    hooked_result = import_data_files(hooked_staging_dataset).collect().with_hooks({import_failed_notification})
 
-    file_metadata_fanout(result, staging_dataset)
-    non_file_metadata_fanout(result, staging_dataset)
+    file_metadata_fanout(result, staging_dataset).with_hooks({import_failed_notification})
+    non_file_metadata_fanout(result, staging_dataset).with_hooks({import_failed_notification}).with_hooks({message_for_import_done})
```

Comment:
🤔 This is fine for now but it got me thinking:

This is exposing a bit of an awkward aspect of this DAG, namely that there is no "final" solid in the pipeline. We're banking on the non_file_metadata_fanout always being the final branch to complete but that might not always be the case. 

I think the real solution here will be to materialize an asset on completion and fire a sensor when such an asset materializes, but that's out of scope for this PR.